Tests/Tests/DatastoreCRUD.m

```diff
@@ -83,6 +83,25 @@ -(NSArray*)generateDocuments:(int)count
 
 #pragma mark - CREATE tests
 
+
+-(void) testDocumentWithInvalidJsonBody
+{
+    NSError * error;
+    CDTMutableDocumentRevision * revision = [CDTMutableDocumentRevision revision];
+    revision.body = @{@"infinity":@(INFINITY)};
+    
+    CDTDocumentRevision * rev = [self.datastore createDocumentFromRevision:revision
+                                                                     error:&error];
+    
+    XCTAssertNil(rev,@"revision is not nil");
+    XCTAssertNotNil(error,@"Error should be set");
+    XCTAssertEqual(400,
+                   error.code,
+                   @"Error code should be 400, but was %ld",
+                   (long)error.code);
+}
```

Comment:
Test name is only sort of accurate. Be more specific, `testDocumentWithInfinityValue`. 

There are other ways the document could be invalid, we should probably test them (e.g., non serialisable object?)